src/custom/pages/Claim/InvestmentFlow/InvestOption.tsx

```diff
@@ -13,78 +12,94 @@ import { ApprovalState } from 'hooks/useApproveCallback'
 import { useCurrencyBalance } from 'state/wallet/hooks'
 import { useActiveWeb3React } from 'hooks/web3'
 import { useClaimDispatchers, useClaimState } from 'state/claim/hooks'
+import { StyledNumericalInput } from 'components/CurrencyInputPanel/CurrencyInputPanelMod'
 
 import { ButtonConfirmed } from 'components/Button'
 import { ButtonSize } from 'theme'
 import Loader from 'components/Loader'
 
-const RangeSteps = styled.div`
-  display: flex;
-  align-items: center;
-  width: 100%;
-  justify-content: space-between;
-`
-
-const RangeStep = styled.button`
-  background: none;
-  border: none;
-  font-size: 0.8rem;
-  cursor: pointer;
-  color: ${({ theme }) => theme.primary1};
-  padding: 0;
-`
-
 const INVESTMENT_STEPS = ['0', '25', '50', '75', '100']
 
-function _scaleValue(maxValue: CurrencyAmount<Currency>, value: string) {
-  // parse percent to string, example 25% -> 4 or 50% -> 2
-  const parsedValue = new Fraction(value, '100')
-
-  // divide maxValue with parsed value to get invest amount
-  return maxValue.multiply(parsedValue).asFraction
+enum ErrorMsgs {
+  Balance = 'Insufficient balance to invest. Adjust the amount or go back to remove this investment option',
+  Input = 'Input amount is bigger then available investment amount',
 }
 
 export default function InvestOption({ approveData, claim, optionIndex }: InvestOptionProps) {
   const { currencyAmount, price, cost: maxCost } = claim
   const { updateInvestAmount } = useClaimDispatchers()
-  const { investFlowData } = useClaimState()
+  const { investFlowData, activeClaimAccount } = useClaimState()
 
-  const investedAmount = useMemo(() => investFlowData[optionIndex].investedAmount, [investFlowData, optionIndex])
+  const { account } = useActiveWeb3React()
 
   const [percentage, setPercentage] = useState<string>(INVESTMENT_STEPS[0])
+  const [typedValue, setTypedValue] = useState<string>('0')
+  const [inputError, setInputError] = useState<string>('')
 
-  const { account } = useActiveWeb3React()
+  const investedAmount = investFlowData[optionIndex].investedAmount
 
   const token = currencyAmount?.currency
-
   const balance = useCurrencyBalance(account || undefined, token)
 
   const decimals = balance?.currency?.decimals
 
-  const handleStepChange = useCallback(
+  const isSelfClaiming = account === activeClaimAccount
+
+  // on invest max amount click handler
+  const setMaxAmount = useCallback(() => {
+    if (!maxCost || !balance) {
+      return
+    }
+
+    const value = maxCost.greaterThan(balance) ? balance : maxCost
+    const amount = value.quotient.toString()
+
+    updateInvestAmount({ index: optionIndex, amount })
+    setTypedValue(formatSmart(value, decimals) || '')
```

Comment:
I don't recommend using `formatSmart` here as it can fuck up with the input by inserting abbreviations such as `<`, `M`, `B`, etc.

See https://github.com/gnosis/cowswap/issues/1330 and https://github.com/gnosis/cowswap/issues/1333

See how that's handled in the current swap input https://github.com/gnosis/cowswap/blob/7099b6f11375724d7fe25143e300f0aedc37b6f6/src/custom/pages/Swap/SwapMod.tsx#L316-L321

Only the opposite field is formatted and as you can see from the issues doesn't work in all cases.